client/wrappers.go

```diff
@@ -0,0 +1,283 @@
+// Needs more focused testing.
+
+package main
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io"
+	"log"
+	"net/http"
+	"os"
+	"strings"
+
+	"github.com/google/uuid"
+)
+
+// BlockingFileUploadTaskData is expected from the add-on.
+type BlockingFileUploadTaskData struct {
+	URL      string `json:"url"`
+	Filepath string `json:"filepath"`
+	AppID    int    `json:"app_id"`
+}
+
+// BlockingFileUploadHandler uploads file to a URL. It is a blocking call by design.
+// It handles the upload of a single file, then returns.
+func BlockingFileUploadHandler(w http.ResponseWriter, r *http.Request) {
```

Comment:
What is the point of all these `Blocking*Handlers`? Why not to send them directly from the add-on?